### PR TITLE
Handle non-source code differently when checking pseudo-assignments

### PR DESCRIPTION
**Pull request by kelloggm**
This prevents warnings from being issued in non-source locations, which are really hard to interpret. Without this, the new case case issues a `required.method.not.called` error for the entire class due to the foreach loop, which is useless to the user.

I don't think there are any generated elements that won't be assigned into a local variable, but I might not be remembering something.

### CODE REVIEW

File: object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java [hunk 1]
```diff
@@ -114,7 +114,6 @@ void checkMustCallInvoked(ControlFlowGraph cfg) {
           new LinkedHashSet<>(curBlockLocals.localSetInfo);
 
       for (Node node : nodes) {
-
```
Comment (msridhar) [inline object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java:117]: remove whitespace change?

File: object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java [hunk 2]
```diff
@@ -717,7 +718,6 @@ private void checkMustCall(
           .noneMatch(localVarTree -> localVarWithTreeSet.contains(localVarTree))) {
         LocalVarWithTree firstlocalVarWithTree = localVarWithTreeSet.iterator().next();
         reportedMustCallErrors.add(firstlocalVarWithTree);
-
```
Comment (msridhar) [inline object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java:720]: remove whitespace change?

**Review by msridhar — APPROVED** [review]
LGTM and makes sense.  Couple optional nits

File: object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java (hunk 2 above)
Comment (kelloggm) [reply object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java:720]: Sorry, didn't notice I had made these (I tried a few different solutions for this before I settled on the one in the PR...).

I kind of prefer the whitespace changes, but I'm happy to revert them if you feel strongly.

Comment (msridhar) [reply object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java:720]: Keeping is ok with me

**Pull request merged**